### ISSUE TITLE
[Java] A false negative about UseIndexOfChar

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3443]
**Affects PMD Version:6.35.0** 

**Rule:UseIndexOfChar**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.36.0/pmd_rules_java_performance.html#useindexofchar>

**Description:**
This is a false negative about the rule UseIndexOfChar. PMD should have reported a warning at line 4. But it doesn't report the warning. Thanks.

**Code Sample demonstrating the issue:**

```java
public void func() {
    String s = "hello world";
    final String var = "d";
    System.out.println(s.indexOf(var)); // Should report a warning here
}
```

**Expected outcome:A warning**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
